### PR TITLE
Login test refactoring

### PR DESCRIPTION
**Pull request by chikodich**
login test refactor, 
test user function implemented, 
new e2e infrastructure implemented

### CODE REVIEW

**Comment by hanny21** [comment]
@chikodich could you please add to `.gitignore` files that are generated by cypress?

```
/cypress/screenshots/
/cypress/videos/
```

File: cypress/integration/login.cy.js [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+  it("user login is unsuccessful with the wrong password and username", () => {
+    LoginSteps.clickStartButton();
+    LoginSteps.fillUsername("notTHErightUSERNAME");
+    LoginSteps.fillPassword("NOTtheRIGHTpassword");
+    LoginSteps.clickLoginButton();
+    LoginAssertions.credNotificationTriggered();
+  })
+
+  it("user login is unsuccessful with the wrong username", () => {
```
Comment (kilimandzsaro) [inline cypress/integration/login.cy.js:46]: Either this, or the previous test is not necessary. Probably the previous test can be removed

File: cypress/integration/login.cy.js [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+  it("user can successfullly logout", () => {
+    LoginSteps.clickStartButton();
+    cy.task("getTestUsername").then((testUsername) => {
+      LoginSteps.fillUsername(testUsername);
+    })
+    LoginSteps.fillPassword(Cypress.config().testDevPassword);
+    LoginSteps.clickLoginButton();
+    LoginSteps.clickLogoutButton();
+    LoginAssertions.loginPageIsLoaded(); 
+    //create cy.logout
```
Comment (kilimandzsaro) [inline cypress/integration/login.cy.js:89]: Do you still need to create the logout? Does this comment needed?

File: cypress/support/assertions/loginAssertions.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
+import LoginPageEls from "../elements/LoginPageEls";
+
+export class LoginAssertions {
+  loginPageIsLoaded() {
+    cy.location("pathname").should("include", "/login");
+  }
+  tokenFieldisLoaded() {
```
Comment (kilimandzsaro) [inline cypress/support/assertions/loginAssertions.js:7]: Can you use a capital "i" for the "is" (proper camel case)

File: cypress/support/assertions/loginAssertions.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    LoginPageEls.tokenField.should("be.visible");//.and("have.text", "/*/") - wildcard in js/cypress;
+  }
+  authenticationSuccessful() {
```
Comment (kilimandzsaro) [inline cypress/support/assertions/loginAssertions.js:10]: Can you add an empty line in front of each method?

File: cypress/support/assertions/loginAssertions.js [hunk 5]
```diff
@@ -0,0 +1,24 @@
+import LoginPageEls from "../elements/LoginPageEls";
+
+export class LoginAssertions {
+  loginPageIsLoaded() {
+    cy.location("pathname").should("include", "/login");
+  }
+  tokenFieldisLoaded() {
+    LoginPageEls.tokenField.should("be.visible");//.and("have.text", "/*/") - wildcard in js/cypress;
```
Comment (kilimandzsaro) [inline cypress/support/assertions/loginAssertions.js:8]: Is the comment necessary?

File: cypress/support/assertions/loginAssertions.js [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+  }
+  authenticationSuccessful() {
+    cy.location("pathname").should("include", "/authenticated");
+  }
+  credNotificationTriggered() {
+    LoginPageEls.notificationText.should("be.visible").and("have.text", "unknown username or password");;
```
Comment (kilimandzsaro) [inline cypress/support/assertions/loginAssertions.js:14]: There's no need for this empty line

File: cypress/support/commands.js [hunk 7]
```diff
@@ -23,3 +26,43 @@
...
+            },
+            "@type": "form",
+            "@id": "00000000-0000-4040-8000-005400000000",
+            "username": username,
+            "password": password,
+        },
+      })
+  }).then((resp) => {
+    expect(resp.body["@type"]).to.contain("verifier");
+    // failure here likely indicates an existing user
```
Comment (kilimandzsaro) [inline cypress/support/commands.js:66]: does this comment necessary?

File: cypress/support/steps/login.js [hunk 8]
```diff
@@ -0,0 +1,29 @@
+import LoginPageEls from "../elements/LoginPageEls";
+
+export class LoginSteps {
+  fillUsername(username) {
+      LoginPageEls.usernameInput.clear({ force: true })
+      LoginPageEls.usernameInput.type(username, { force: true });
+    return this;
+  }
+  fillPassword(password) {
```
Comment (kilimandzsaro) [inline cypress/support/steps/login.js:9]: Please add the empty lines between the methods everywhere

File: package.json [hunk 9]
```diff
@@ -19,6 +19,8 @@
   },
   "devDependencies": {
     "cypress": "^10.0.3",
+    "cypress-dotenv": "^2.0.0",
```
Comment (kilimandzsaro) [inline package.json:22]: I don't think this package is necessary anymore. You are using the built-in Cypress.env()

File: package.json [hunk 10]
```diff
@@ -19,6 +19,8 @@
...
+    "dotenv": "^16.0.2",
```
Comment (kilimandzsaro) [inline package.json:23]: Same as `cypress-dotenv`, I don't think this is necessary anymore

File: packages/web-ts/package.json [hunk 11]
```diff
@@ -35,6 +35,8 @@
     "@types/react": "^18.0.12",
     "@types/react-dom": "^18.0.5",
     "@types/react-router-dom": "^5.3.3",
+    "cypress-dotenv": "^2.0.0",
```
Comment (kilimandzsaro) [inline packages/web-ts/package.json:38]: This is not needed here

File: packages/web-ts/package.json [hunk 12]
```diff
@@ -35,6 +35,8 @@
...
+    "dotenv": "^16.0.2",
```
Comment (kilimandzsaro) [inline packages/web-ts/package.json:39]: This package is not needed here

**Review by kilimandzsaro — CHANGES_REQUESTED** [review]

File: cypress/integration/login.cy.js (hunk 1 above)
Comment (chikodich) [reply cypress/integration/login.cy.js:46]: "user login is unsuccessful with the wrong password and username" - removed

File: cypress/integration/login.cy.js (hunk 2 above)
Comment (chikodich) [reply cypress/integration/login.cy.js:89]: this should say cy.Command.logout - will remove and implement later

File: cypress/support/assertions/loginAssertions.js (hunk 3 above)
Comment (chikodich) [reply cypress/support/assertions/loginAssertions.js:7]: updated

File: cypress/support/assertions/loginAssertions.js (hunk 4 above)
Comment (chikodich) [reply cypress/support/assertions/loginAssertions.js:10]: updated

File: cypress/support/assertions/loginAssertions.js (hunk 5 above)
Comment (chikodich) [reply cypress/support/assertions/loginAssertions.js:8]: was a note to self for future - removed

File: cypress/support/commands.js (hunk 7 above)
Comment (chikodich) [reply cypress/support/commands.js:65]: comment removed

File: cypress/support/steps/login.js (hunk 8 above)
Comment (chikodich) [reply cypress/support/steps/login.js:9]: updated

File: package.json (hunk 9 above)
Comment (chikodich) [reply package.json:22]: updated

File: packages/web-ts/package.json (hunk 11 above)
Comment (chikodich) [reply packages/web-ts/package.json:38]: updated

File: packages/web-ts/package.json (hunk 12 above)
Comment (chikodich) [reply packages/web-ts/package.json:39]: updated

**Review by kilimandzsaro — APPROVED** [review]

**Pull request merged**